### ISSUE TITLE
[Bug]: Auto update new courses in autocomplete graph

### ISSUE THREAD

**Issue by jpahm** [UTDNebula/utd-trends#270]
### What did you expect to happen?

The ability to search for new courses from the current (24f) semester, i.e. ECS 2390.

### What happened?

The course cannot be searched and isn't provided via autocomplete despite the course data being present in the API.

### Version

Beta 0.1.0

### What browsers are you seeing the problem on?

Firefox

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jpahm** [comment]
Note that a good long-term solution for this sort of issue will probably involve setting up a system to pull the latest DAG data from the API's `autocomplete` endpoint and generating the associated autocomplete graphs.

**Comment by TyHil** [comment]
Ah I didn't know the autocomplete endpoint was functional, I'll work on that

**Comment by TyHil** [comment]
On hold until the API endpoint is deployed

**Comment by AbhiramTadepalli** [comment]
@TyHil when someone searches for ECS 2390, should we display ECS 3390 distribution data?

**Comment by TyHil** [comment]
Idk, might be a little misleading to display 3390 data for 2390 since there have probably been some class changes. Hopefully people would know to look up 3390, but the fact they have to do that would indicate it is different from 2390.

**Comment by TyHil** [comment]
This issue is blocked by [nebula-api/#225](https://github.com/UTDNebula/nebula-api/issues/225)

> @TyHil when someone searches for ECS 2390, should we display ECS 3390 distribution data? and @jpahm when we call for ECS2390 grade distribution data, does API provide 3390 data from past semesters (also is there 2390 grade data yet)?

This is blocked by [nebula-api/#223](https://github.com/UTDNebula/nebula-api/issues/223)

**Comment by AbhiramTadepalli** [comment]
> Idk, might be a little misleading to display 3390 data for 2390 since there have probably been some class changes. Hopefully people would know to look up 3390, but the fact they have to do that would indicate it is different from 2390.

When https://github.com/UTDNebula/nebula-api/issues/223 is completed, it will in fact group ECS 2390 and ECS 3390.

Now, _we_ could detect the change maybe and let users know the course name has changed? That requires that the course API also groups by `internal_course_number`

**Comment by TyHil** [comment]
If API is doing the grouping, seems reasonable to just do that then

**Comment by mikehquan19** [comment]
Hi! I'm the one in charge of https://github.com/UTDNebula/nebula-api/issues/223. The task will be completed by the end of Friday at the latest, and the issue with grouping will be resolved.

**Comment by jpahm** [comment]
We could probably look into making the grouping optional if such functionality would be preferred.

**Comment by jpahm** [comment]
> This issue is blocked by [nebula-api/#225](https://github.com/UTDNebula/nebula-api/issues/225)
> 
> > @TyHil when someone searches for ECS 2390, should we display ECS 3390 distribution data? and @jpahm when we call for ECS2390 grade distribution data, does API provide 3390 data from past semesters (also is there 2390 grade data yet)?
> 
> This is blocked by [nebula-api/#223](https://github.com/UTDNebula/nebula-api/issues/223)

Both of these are now unblocked via the latest API deployment :)
